### PR TITLE
Add context to contentUpdatedAt field at statistics

### PR DESCRIPTION
**Pull request by y-korneev**

### CODE REVIEW

File: xyz-hub-service/src/main/java/com/here/xyz/hub/task/FeatureTaskHandler.java [hunk 1]
```diff
@@ -1543,11 +1545,41 @@ private static void defineGlobalSearchableField(StatisticsResponse response, Fea
...
-  private static void defineContentUpdatedAtField(StatisticsResponse response, FeatureTask task) {
-    StatisticsResponse.Value<Long> contentUpdatedAtVal = new StatisticsResponse.Value(task.space.contentUpdatedAt);
-    // Due to caching the value of contentUpdatedAt field could be obsolete in some edge cases
-    contentUpdatedAtVal.setEstimated(true);
-    response.setContentUpdatedAt(contentUpdatedAtVal);
+  private static Future<Void> defineContentUpdatedAtField(StatisticsResponse response, FeatureTask.GetStatistics task) {
+    Promise<Void> p = Promise.promise();
+
+    if (task.space.getExtension() != null) {
+      MultiMap queryParams = task.context.get("queryParams");
```
Comment (roegi) [inline xyz-hub-service/src/main/java/com/here/xyz/hub/task/FeatureTaskHandler.java:1552]: Reading / parsing of query params happened in REST tier already. No need to parse the query params on task level again. Instead we could read the context like: `task.getEvent().getContext()`

File: xyz-hub-service/src/main/java/com/here/xyz/hub/task/FeatureTaskHandler.java [hunk 2]
```diff
@@ -1543,11 +1545,41 @@ private static void defineGlobalSearchableField(StatisticsResponse response, Fea
...
+              .filter(entry -> entry.getKey().equalsIgnoreCase("context"))
+              .map(entry -> entry.getValue().toUpperCase()).findFirst().orElse(null);
+      ContextAwareEvent.SpaceContext spaceContext = ContextAwareEvent.SpaceContext.of(spaceContextStr);
+      Space.resolveSpace(task.getMarker(), task.space.getExtension().getSpaceId())
+              .onSuccess(space -> {
+                long contentUpdatedAt;
+                if (spaceContext == ContextAwareEvent.SpaceContext.SUPER) {
+                  contentUpdatedAt = space.contentUpdatedAt;
+                } else if (spaceContext == ContextAwareEvent.SpaceContext.DEFAULT) {
+                  contentUpdatedAt = space.contentUpdatedAt > task.space.contentUpdatedAt
```
Comment (roegi) [inline xyz-hub-service/src/main/java/com/here/xyz/hub/task/FeatureTaskHandler.java:1563]: Could be simplified to: `Math.max(space.contentUpdatedAt, task.space.contentUpdatedAt)`

File: xyz-hub-service/src/main/java/com/here/xyz/hub/task/FeatureTaskHandler.java [hunk 3]
```diff
@@ -1543,11 +1545,41 @@ private static void defineGlobalSearchableField(StatisticsResponse response, Fea
...
+      String spaceContextStr = queryParams.entries().stream()
+              .filter(entry -> entry.getKey().equalsIgnoreCase("context"))
+              .map(entry -> entry.getValue().toUpperCase()).findFirst().orElse(null);
+      ContextAwareEvent.SpaceContext spaceContext = ContextAwareEvent.SpaceContext.of(spaceContextStr);
+      Space.resolveSpace(task.getMarker(), task.space.getExtension().getSpaceId())
+              .onSuccess(space -> {
+                long contentUpdatedAt;
+                if (spaceContext == ContextAwareEvent.SpaceContext.SUPER) {
+                  contentUpdatedAt = space.contentUpdatedAt;
+                } else if (spaceContext == ContextAwareEvent.SpaceContext.DEFAULT) {
```
Comment (roegi) [inline xyz-hub-service/src/main/java/com/here/xyz/hub/task/FeatureTaskHandler.java:1562]: Could be simplified to `DEFAULT`

File: xyz-hub-service/src/main/java/com/here/xyz/hub/task/FeatureTaskHandler.java [hunk 4]
```diff
@@ -1543,11 +1545,41 @@ private static void defineGlobalSearchableField(StatisticsResponse response, Fea
...
+    if (task.space.getExtension() != null) {
+      MultiMap queryParams = task.context.get("queryParams");
+      String spaceContextStr = queryParams.entries().stream()
+              .filter(entry -> entry.getKey().equalsIgnoreCase("context"))
+              .map(entry -> entry.getValue().toUpperCase()).findFirst().orElse(null);
+      ContextAwareEvent.SpaceContext spaceContext = ContextAwareEvent.SpaceContext.of(spaceContextStr);
+      Space.resolveSpace(task.getMarker(), task.space.getExtension().getSpaceId())
+              .onSuccess(space -> {
+                long contentUpdatedAt;
+                if (spaceContext == ContextAwareEvent.SpaceContext.SUPER) {
```
Comment (roegi) [inline xyz-hub-service/src/main/java/com/here/xyz/hub/task/FeatureTaskHandler.java:1560]: Could be simplified to `SUPER`

File: xyz-hub-service/src/main/java/com/here/xyz/hub/task/FeatureTaskHandler.java [hunk 5]
```diff
@@ -1543,11 +1545,41 @@ private static void defineGlobalSearchableField(StatisticsResponse response, Fea
...
+                }
+                StatisticsResponse.Value<Long> contentUpdatedAtVal = new StatisticsResponse.Value(contentUpdatedAt);
+                // Due to caching the value of contentUpdatedAt field could be obsolete in some edge cases
+                contentUpdatedAtVal.setEstimated(true);
+                response.setContentUpdatedAt(contentUpdatedAtVal);
+                p.complete();
+              })
+              .onFailure(t -> p.fail(t));
+    } else {
+      StatisticsResponse.Value<Long> contentUpdatedAtVal = new StatisticsResponse.Value(task.space.contentUpdatedAt);
```
Comment (roegi) [inline xyz-hub-service/src/main/java/com/here/xyz/hub/task/FeatureTaskHandler.java:1576]: Duplicated code (see: lines: 1568 ff)

File: xyz-hub-service/src/main/java/com/here/xyz/hub/task/FeatureTaskHandler.java (hunk 1 above)
Comment (y-korneev) [reply xyz-hub-service/src/main/java/com/here/xyz/hub/task/FeatureTaskHandler.java]: Fixed, thanks.

File: xyz-hub-service/src/main/java/com/here/xyz/hub/task/FeatureTaskHandler.java (hunk 5 above)
Comment (y-korneev) [reply xyz-hub-service/src/main/java/com/here/xyz/hub/task/FeatureTaskHandler.java]: Simplified duplicated code to one line

File: xyz-hub-service/src/main/java/com/here/xyz/hub/task/FeatureTaskHandler.java (hunk 3 above)
Comment (y-korneev) [reply xyz-hub-service/src/main/java/com/here/xyz/hub/task/FeatureTaskHandler.java]: Done

**Review by roegi — APPROVED** [review]

**Pull request merged**